src/client/containers/layout.js

```diff
@@ -42,9 +42,25 @@ const {Alert, MenuItem, Nav, Navbar, NavItem, NavDropdown} = bootstrap;
 class Layout extends React.Component {
 	constructor(props) {
 		super(props);
-		this.state = {};
+		this.state = {keepMenuOpen: false, menuOpen: false};
 		this.renderNavContent = this.renderNavContent.bind(this);
 		this.renderNavHeader = this.renderNavHeader.bind(this);
+		this.handleDropdownToggle = this.handleDropdownToggle.bind(this);
+		this.handleDropdownClick = this.handleDropdownClick.bind(this);
+	}
+
+	handleDropdownToggle(newValue) {
+		if (this.state.keepMenuOpen) {
+			this.setState({menuOpen: true});
+			this.setState({keepMenuOpen: false});
```

Comment:
`setState` takes an object with the variables you want to change. That object can have multiple keys, so you can set both variables in one go:
`this.setState({menuOpen: true, keepMenuOpen: false});`

Reply:
Done